File: 12-pendahuluan inheritance/main.php
<?php 

// CONTOH DI PHP

class Hero 
{
    public $name;
    public $health;
    public function __construct($name, $health) 
    {
        $this->name = $name;
        $this->health = $health;
    }
};

class HeroFighter extends Hero
{
    public function __construct($name, $health) 
    {
        parent::__construct($name, $health);
    }
};

$alucard = new HeroFighter("alucard", 1000);

var_dump([
    'name' => $alucard->name,
    'health' => $alucard->health,
]);

File: 12-pendahuluan inheritance/main.py
class Hero :
    def __init__(self, name, health):
        self.name = name
        self.health = health

class HeroIntelegent(Hero) :
    pass

class HeroStrength(Hero) :
    pass

lina = Hero('lina', 100)
techies = HeroIntelegent('techies', 50)
axe = HeroStrength('axe', 50)

print(lina.name)
print(techies.name)
print(axe.name)